engine/lib/pagehandler.php

```diff
@@ -58,6 +58,101 @@ function elgg_generate_url($name, array $parameters = []) {
 	return _elgg_services()->router->generateUrl($name, $parameters);
 }
 
+/**
+ * Generate entity URL from a named route
+ * Route segments will be automatically resolved from entity attributes and metadata
+ *
+ * @param ElggEntity $entity      Entity
+ * @param string     $resource    Resource name
+ * @param string     $subresource Subresource name
+ * @param array      $query       URL query elements
```

Comment:
Rather this be $params. $query makes me think querystring.

Reply:
Anything that's not defined as a URL segment goes into query string

Reply:
I will rename and update the docs for both